File: algorithms/0392.is-subsequence/solution.go
func isSubsequence(s string, t string) bool {
	letters := []rune(t)

	for _, r := range s {
		i := indexOf(r, letters)
		if i >= 0 {
			letters = letters[i+1:]
		} else {
			return false
		}
	}

	return true
}

func indexOf(target rune, runes []rune) int {
	for i, r := range runes {
		if r == target {
			return i
		}
	}
	return -1
}

File: algorithms/0392.is-subsequence/solution.rb
# @param {String} s
# @param {String} t
# @return {Boolean}
def is_subsequence(s, t)
  letters = t.chars

  s.chars do |c|
    idx = letters.index(c)
    if idx
      letters = letters[idx + 1..]
    else
      return false
    end
  end

  true
end
